scripts/zones/RoMaeve/npcs/Goblin_Footprint.lua

```diff
@@ -0,0 +1,66 @@
+-----------------------------------
+-- Area: Bostaunieux Oubliette
+--  NPC: Goblin Footprint
+-- Type: NPC
+-- !pos  111.806 -24.007 57.266 167
+-----------------------------------
+local ID = require("scripts/zones/RoMaeve/IDs")
+require("scripts/globals/pasteventwatcher")
+require("scripts/globals/quests")
+require("scripts/globals/missions")
+-----------------------------------
+---
+function onTrade(player,npc,trade)
+    local options = {}
+    if player:hasCompletedMission(WINDURST,dsp.mission.id.windurst.VAIN) == true then
+        options[1] = 1
+    end
+    if player:hasCompletedMission(WINDURST,dsp.mission.id.windurst.MOON_READING) == true then
+        options[2] = 1
+    end
+    if player:getQuestStatus(ZILART,dsp.quest.id.outlands.DIVINE_MIGHT) == QUEST_COMPLETED then
+        options[3] = 1
+    end
+    if player:hasCompletedMission(ASA,dsp.mission.id.asa.FOUNTAIN_OF_TROUBLE) == true then
+        options[4] = 1
+    end
+    local eventList = dsp.pasteventwatcher.goblinFootprint({options[1], options[2], options[3], options[4]})
+    if not next(options) == nil then
+        player:startEvent(6, eventList, 4294967294, 4294967294, 4294967294)
```

Comment:
This looks like you're hardcoding garbage data from a packet capture. Remember that params are not zeroed out for every packet on retail. You may be getting bad data for unused params based on what packet was sent prior to this one.

Reply:
Hi @nfaletra  thanks for taking time on reviewing my PR 😸

The three parameters of 0xFFFFFFFE are needed to be included for the goblins else it won't populate the table. I'm attaching 3 packets for gobbies in Zilart and Base game area. (Thanks to @UynGH 's hard work)
<img width="531" alt="gobbie" src="https://user-images.githubusercontent.com/26943220/66094295-53550e80-e548-11e9-85fb-1b87db349ea2.PNG">
<img width="815" alt="gobbie3" src="https://user-images.githubusercontent.com/26943220/66094296-53550e80-e548-11e9-8bed-c50a3c718cd1.PNG">
<img width="535" alt="gobbie2" src="https://user-images.githubusercontent.com/26943220/66094297-53550e80-e548-11e9-8520-6759cad169aa.PNG">

Reply:
Does it need to be 0xFFFFFFFE, or just a certain bit that the currently used param happens to have set? (Ie: Is it really something like 255, or just 2?).

Reply:
> Also, do we have a hard "no hex in params" rule? 

hex is actually preferred ***if*** its as a bitmask in the script. we de-hexed the csid themselves though.